### PR TITLE
StyleCop configuration

### PR DESCRIPTION
**Pull request by bdukes**
First follow-up to #3802 

I've added the default configuration inside `stylecop.json`, I'm going to propose some changes from the defaults as PR review comments, and we can try to come to a consensus.  There is also the option of [using a `ruleset` file to turn some rules off altogether](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/Configuration.md#sharing-configuration-among-solutions).

I've also added an `.editorconfig` file, which is a mixture of the [`.editorconfig` file from StyleCop](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/.editorconfig) and the [`.editorconfig` files from `dotnet-format`](https://github.com/dotnet/format/blob/master/.editorconfig), with StyleCop "winning" where they conflict.  Some of the notable differences:
* Always use `this` 
* No prefix for static and instance fields (the `dotnet-format` style uses `s_` for static fields and `_` for fields)
* Prefer `var` only where type is obvious (the `dotnet-format` style prefers `var` everywhere)
* Require braces (the `dotnet-format` style prefers but does not enforce braces)

### CODE REVIEW

File: stylecop.json [hunk 1]
```diff
@@ -1,10 +1,49 @@
...
+      "allowedHungarianPrefixes": [],
+      "allowedNamespaceComponents": [],
+      "includeInferredTupleElementNames": false,
+      "tupleElementNameCasing": "PascalCase"
+    },
+    "maintainabilityRules": {
+      "topLevelTypes": ["class"]
+    },
+    "layoutRules": {
+      "newlineAtEndOfFile": "allow",
```
Comment (bdukes) [inline stylecop.json:32]:
```suggestion
      "newlineAtEndOfFile": "require",
```

File: stylecop.json [hunk 2]
```diff
@@ -1,10 +1,49 @@
...
+      "elementOrder": [
+        "kind",
+        "accessibility",
+        "constant",
+        "static",
+        "readonly"
+      ],
+      "systemUsingDirectivesFirst": true,
+      "usingDirectivesPlacement": "insideNamespace",
+      "blankLinesBetweenUsingGroups": "allow"
```
Comment (bdukes) [inline stylecop.json:19]:
```suggestion
      "blankLinesBetweenUsingGroups": "require"
```

File: stylecop.json (hunk 1 above)
Comment (david-poindexter) [reply stylecop.json:32]: Why should this be required?  Just for consistency's sake?

File: stylecop.json (hunk 2 above)
Comment (david-poindexter) [reply stylecop.json:19]: I agree with this one - it will certainly help with readability.

File: stylecop.json (hunk 1 above)
Comment (bdukes) [reply stylecop.json:32]: Yes.  Also if it's required then we can just auto-fix it and be done.

Comment (david-poindexter) [reply stylecop.json:32]: Sounds like a good plan - thanks @bdukes - I'll go ahead and commit these suggestions.

**Pull request merged**